state-transition/core/state_processor_exits.go

```diff
@@ -29,49 +29,43 @@ import (
 
 // InitiateValidatorExit initiates the exit of the validator with index `idx`.
 func (sp *StateProcessor) InitiateValidatorExit(st *statedb.StateDB, idx math.ValidatorIndex) error {
+	// Return if the validator already initiated an exit, making sure to only exit validators once.
 	validator, err := st.ValidatorByIndex(idx)
 	if err != nil {
 		return err
 	}
-	// By this point, the state should be correctly updated with the fork.
+	if validator.GetExitEpoch() != constants.FarFutureEpoch {
+		return nil
+	}
```

Comment:
noting this check is in spec (pre-electra) and will not affect backwards compat for us, so moving to outside of fork version check

Reply:
https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#initiate-validator-exit

Reply:
removing this as it changes logic. dont want this PR to change logic.